### PR TITLE
Fix typos discovered by codespell in /docs

### PR DESCRIPTION
**Pull request by cclauss**
[`codespell --ignore-words-list="hel,wile" --skip="*.css,*.js,*.po" ./docs`](https://pypi.org/project/codespell)

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
It all looks good to me! Thanks a lot @cclauss . We should have this in CI.

File: docs/guides/searching-with-readthedocs.rst [hunk 1]
```diff
@@ -80,7 +80,7 @@ Example queries:
 
 - https://docs.readthedocs.io/?rtd_search=reedthedcs~2
 - https://docs.readthedocs.io/?rtd_search=authentation~3
-- https://docs.readthedocs.io/?rtd_search=configurtion~1
+- https://docs.readthedocs.io/?rtd_search=configuration~1
```
Comment (ericholscher) [inline docs/guides/searching-with-readthedocs.rst:83]: I think this was actually meant to be mispelled :)

**Comment by ericholscher** [comment]
This is a great PR, thanks! :)

File: docs/guides/searching-with-readthedocs.rst (hunk 1 above)
Comment (astrojuanlu) [reply docs/guides/searching-with-readthedocs.rst:83]: Woops, didn't notice that one, sorry

Comment (cclauss) [reply docs/guides/searching-with-readthedocs.rst:83]: Reverting in #8419

**Pull request merged**